std/assembly/reference.ts

```diff
@@ -20,6 +20,12 @@ export abstract class Eqref extends Ref {
 
 @final @unmanaged
 export abstract class I31ref extends Ref {
+  get(this: i31ref): i32 {
+    return i31.get(this);
+  }
+  getUnsigned(this: i31ref): u32 {
+    return <u32>i31.get(this);
+  }
```

Comment:
I think it's unnecessary. It's much shorted use `<u32>ref.get()` instead `ref.getUnsigned()` I guess

Reply:
The problem there is that an `i31` does not properly sign-extend, so it is important how it is loaded. Say we are loading `0x7fffffff`, then loading that as signed becomes `-1`, but `<u32>-1` gives us `0xffffffff`, not `0x7fffffff`, as would have been the value when loading the `i31` as unsigned.

Reply:
I see. Hmm

What about:
```ts
class I31ref {
  signed(this: i31ref): i32 {
    return i31.get<i32>(this);
  }
  
  unsigned(this: i31ref): u32 {
    return i31.get<u32>(this);
  }
}
```
or
```ts
class I31ref {
  i32(this: i31ref): i32 {
    return i31.get<i32>(this);
  }
  
  u32(this: i31ref): u32 {
    return i31.get<u32>(this);
  }
}
```

Reply:
Wondering, what if we do this as casts instead:

```ts
var ref = i31.new(42);
var i32 = <i32>ref;
var u32 = <u32>ref;
```

Reply:
yeah. that's will be the best way